### ISSUE TITLE
Confirm delete

### ISSUE THREAD

**Issue by katanaur** [felixwiemuth/SimpleReminder#25]
When Long-click on a reminder in a list to select it the available actions are shown in upper right corner. Symbol for deleting reminder is just next to the options button (3 dots) and it could easily be tapped by accident. Since there is no confirmation dialog for delete action reminder could be lost by mistake.
So, delete symbol should be moved away from option symbol, and confirmation dialog for delete would be great anyway.
Also, maybe to add trash in a reminder list, so that all deleted items go there, with additional option to empty trash.

By the way, great little app.

**Comment by felixwiemuth** [comment]
I agree that it is not optimal right now, and I have also considered solutions before.

When templates are available, it will also be important to have a similar safety mechanism for those.

I can think of several possible solutions.

### Undo (easiest to implement)

Every delete action is remembered (it can be the deletion of just one reminder or a whole selection) and the main options menu gets an extra entry "Undo delete x reminders" (where x is the number of reminders deleted last).

When clicking on the undo option in the menu, the last delete action will be reverted by adding the deleted reminders again to the list of reminders and the menu entry will be updated to represent the delete action before this (if any).

The deleted reminders cannot be seen though, and undo might only be available until the app is left (this would make implementation easier).

In addition, just after deleting, a toast popup could include an "Undo" button to undo the action directly.

### Trash (preferred solution)

Deleted reminders are moved to a trash.

The trash could be a special section at the end of the reminders list. Here entries can be selected as other reminders, but instead with the options to restore or delete finally.

An "Empty trash" option could clear the trash.

This variant has the advantage that with an "auto delete" feature (delete done reminders older than a certain amount of days) reminders could be moved to trash rather than be deleted directly.

Similarly, an option to trash reminders instead of adding them to the list of done reminders when dismissing a notification would be less risky than a delete-directly version.

The "undo" toast popup would still be an option in this variant.

### Confirmation dialog (probably not preferred)
I think the above solutions are superior to a confirmation dialog, especially as accidentally deleting should be the exception, so that in most cased it would just be annoying to have to confirm to delete.

Also, a confirmation dialog is not an option when reminders are deleted automatically.

### Behaviour when restoring reminders

Undoing or restoring from trash should probably reschedule reminders that at the time of deletion had status "scheduled" (and show those immediately which became due in the meantime) and put the remaining ones under "Done".
